lib/elements/dom-repeat.html

```diff
@@ -687,8 +687,8 @@
      * If `sort` is provided, the index will reflect the sorted order (rather
      * than the original array order).
      *
-     * @param {HTMLElement} el Element for which to return the index.
-     * @return {*} Row index associated with the element (note this may
+     * @param {!HTMLElement} el Element for which to return the index.
+     * @return {number} Row index associated with the element (note this may
```

Comment:
It looks like this could also return `null`, see the short circuit evaluation where `modelForElement` can return `null`.